File: leetcode_js/src/plusOne.js
const plusOne = function (digits) {
  const result = [];
  let extra = 0;

  for (let i = digits.length - 1; i >= 0; i--) {
    if (i === digits.length - 1) {
      const add = digits[i] + 1;

      if (add > 9) {
        result.unshift(add - 10);
        extra = 1;
      } else {
        result.unshift(add);
        extra = 0;
      }
    } else {
      const add = digits[i] + extra;

      if (add > 9) {
        result.unshift(add - 10);
        extra = 1;
      } else {
        result.unshift(add);
        extra = 0;
      }
    }
  }

  if (extra === 1) {
    result.unshift(1);
  }

  return result;
};

module.exports = { plusOne };

File: leetcode_python/lib/problems/p0066_plus_one/plus_one.py
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        result = []
        extra = 0

        for i in range(len(digits) - 1, -1, -1):
            if i == len(digits) - 1:
                n = digits[i] + 1

                if n >= 10:
                    n -= 10
                    extra = 1

                result.append(n)
            else:
                n = digits[i] + extra
                extra = 0

                if n >= 10:
                    n -= 10
                    extra = 1

                result.append(n)

        if extra > 0:
            result.append(1)

        result.reverse()

        return result
